presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java

```diff
@@ -236,7 +261,24 @@ private long getRowCount(TpchTableHandle tpchTableHandle)
             case "customer":
                 return (long) (150_000 * scaleFactor);
             case "orders":
-                return (long) (1_500_000 * scaleFactor);
+                Set<String> orderStatusValues = predicate.map(tupleDomain ->
+                        ORDER_STATUS_NULLABLE_VALUES.stream()
+                                .filter(convertToPredicate(tupleDomain, OrderColumn.ORDER_STATUS))
+                                .map(nullableValue -> ((Slice) nullableValue.getValue()).toStringUtf8())
+                                .collect(toSet()))
+                        .orElse(ORDER_STATUS_VALUES);
+
+                long totalRows = 0L;
+                if (orderStatusValues.contains("F")) {
+                    totalRows = 729_413;
```

Comment:
it's a base value for 1Gb which should stay constant since TPCH data layout is predictable. This could be moved to static final. Other than that, I don't know how to improve it right now. As @losipiuk mentioned, we could somehow precompute those beforehand, but it's not trivial either.